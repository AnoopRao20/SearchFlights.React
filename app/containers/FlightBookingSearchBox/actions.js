/*
 *
 * FlightBookingSearchBox actions
 *
 */

import {
  RESET_STATE,
  LOAD_SOURCE_CITIES,
  LOAD_SOURCE_CITIES_SUCCESS,
  LOAD_SOURCE_CITIES_FAIL,
  LOAD_DESTINATION_CITIES,
  LOAD_DESTINATION_CITIES_SUCCESS,
  LOAD_DESTINATION_CITIES_FAIL,
  SOURCE_CITY_CHANGE,
  DESTINATION_CITY_CHANGE,
  TRAVEL_DATE_CHANGE,
  RETURN_DATE_CHANGE,
  PERFORM_SEARCH,
} from './constants';

export function resetState() {
  return {
    type: RESET_STATE,
  };
}

export function loadSourceCities(name) {
  return {
    type: LOAD_SOURCE_CITIES,
    name,
  };
}

export function loadSourceCitiesSuccess(cities) {
  return {
    type: LOAD_SOURCE_CITIES_SUCCESS,
    cities,
  };
}

export function loadSourceCitiesFail() {
  return {
    type: LOAD_SOURCE_CITIES_FAIL,
  };
}

export function loadDestinationCities(name) {
  return {
    type: LOAD_DESTINATION_CITIES,
    name,
  };
}

export function loadDestinationCitiesSuccess(cities) {
  return {
    type: LOAD_DESTINATION_CITIES_SUCCESS,
    cities,
  };
}

export function loadDestinationCitiesFail() {
  return {
    type: LOAD_DESTINATION_CITIES_FAIL,
  };
}

export function sourceCityChange(sourceCity) {
  return {
    type: SOURCE_CITY_CHANGE,
    sourceCity,
  };
}

export function destinationCityChange(destinationCity) {
  return {
    type: DESTINATION_CITY_CHANGE,
    destinationCity,
  };
}

export function travelDateChange(travelDate) {
  return {
    type: TRAVEL_DATE_CHANGE,
    travelDate,
  };
}

export function returnDateChange(returnDate) {
  return {
    type: RETURN_DATE_CHANGE,
    returnDate,
  };
}

export function performSearch() {
  return {
    type: PERFORM_SEARCH,
  };
}
