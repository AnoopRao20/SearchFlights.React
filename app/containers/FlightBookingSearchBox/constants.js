/*
 *
 * FlightBookingSearchBox constants
 *
 */

const scope = 'app/FlightBookingSearchBox/';

export const RESET_STATE = `${scope}RESET_STATE`;

export const LOAD_SOURCE_CITIES = `${scope}LOAD_SOURCE_CITIES`;
export const LOAD_SOURCE_CITIES_SUCCESS = `${scope}LOAD_SOURCE_CITIES_SUCCESS`;
export const LOAD_SOURCE_CITIES_FAIL = `${scope}LOAD_SOURCE_CITIES_FAIL`;

export const LOAD_DESTINATION_CITIES = `${scope}LOAD_DESTINATION_CITIES`;
export const LOAD_DESTINATION_CITIES_SUCCESS = `${scope}LOAD_DESTINATION_CITIES_SUCCESS`;
export const LOAD_DESTINATION_CITIES_FAIL = `${scope}LOAD_DESTINATION_CITIES_FAIL`;

export const SOURCE_CITY_CHANGE = `${scope}SOURCE_CITY_CHANGE`;
export const DESTINATION_CITY_CHANGE = `${scope}DESTINATION_CITY_CHANGE`;
export const TRAVEL_DATE_CHANGE = `${scope}TRAVEL_DATE_CHANGE`;
export const RETURN_DATE_CHANGE = `${scope}RETURN_DATE_CHANGE`;

export const PERFORM_SEARCH = `${scope}PERFORM_SEARCH`;
