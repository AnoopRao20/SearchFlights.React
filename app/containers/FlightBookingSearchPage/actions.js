/*
 *
 * FlightBookingSearchPage actions
 *
 */

import {
  RESET_STATE,
  LOAD_DATA,
  LOAD_DATA_SUCCESS,
  LOAD_DATA_FAIL,
} from './constants';

export function resetState() {
  return {
    type: RESET_STATE,
  };
}

export function loadData() {
  return {
    type: LOAD_DATA,
  };
}

export function loadDataSuccess(tableData) {
  return {
    type: LOAD_DATA_SUCCESS,
    tableData,
  };
}

export function loadDataFail() {
  return {
    type: LOAD_DATA_FAIL,
  };
}
