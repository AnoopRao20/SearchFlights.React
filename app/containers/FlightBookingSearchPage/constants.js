/*
 *
 * FlightBookingSearchPage constants
 *
 */

const scope = 'app/FlightBookingSearchPage/';

export const RESET_STATE = `${scope}RESET_STATE`;

export const LOAD_DATA = `${scope}LOAD_DATA`;
export const LOAD_DATA_SUCCESS = `${scope}LOAD_DATA_SUCCESS`;
export const LOAD_DATA_FAIL = `${scope}LOAD_DATA_FAIL`;
