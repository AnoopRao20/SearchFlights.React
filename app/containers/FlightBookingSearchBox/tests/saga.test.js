/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put, takeLatest } from 'redux-saga/effects';
import { LOAD_SOURCE_CITIES } from '../constants';
import { loadSourceCitiesSuccess, loadSourceCitiesFail } from '../actions';

import flightBookingSearchBoxSaga, { loadSourceCities } from '../saga';

const name = 'Jaipur';

describe('loadSourceCities Saga', () => {
  let getSourceCityGenerator;

  beforeEach(() => {
    getSourceCityGenerator = loadSourceCities(name);

    const callDescriptor = getSourceCityGenerator.next(name).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the loadSourceCitiesSuccess action if it requests the data successfully', () => {
    const response = [
      {
        id: '9',
        name: 'Jaipur',
        state: 'Rajasthan',
      },
    ];
    const putDescriptor = getSourceCityGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(loadSourceCitiesSuccess(response)));
  });

  it('should call the loadSourceCitiesFail action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getSourceCityGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(loadSourceCitiesFail()));
  });
});

describe('flightBookingSearchBoxSaga Saga', () => {
  const sourceCityDataSaga = flightBookingSearchBoxSaga();

  it('should start task to watch for LOAD_SOURCE_CITIES action', () => {
    const takeLatestDescriptor = sourceCityDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(LOAD_SOURCE_CITIES, loadSourceCities),
    );
  });
});
