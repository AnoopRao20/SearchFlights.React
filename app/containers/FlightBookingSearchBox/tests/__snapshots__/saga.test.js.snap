// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSourceCities Saga should call the loadSourceCitiesFail action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`loadSourceCities Saga should dispatch the loadSourceCitiesSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
